src/framework/anim/controller/anim-controller.js

```diff
@@ -516,8 +521,17 @@ class AnimController {
         let state;
         let animation;
         let clip;
-        this._timeInStateBefore = this._timeInState;
-        this._timeInState += dt * this.activeState.speed;
+        // update time when looping or when the active state is not at the end of its duration
+        if (this.activeState.loop || this._timeInState < this.activeStateDuration) {
```

Comment:
`this.activeStateDuration` is really expensive - it searches for animations by name, and it gets called here 4x ?

Reply:
I've added a flag to the calculation of the current states duration. It'll now be calculated at most once per state transition or assignment of a new animation file.